services/alertmanager/alertmanager.go

```diff
@@ -181,6 +195,68 @@ func (svc *Service) updateConfiguration(ctx context.Context) {
 	svc.l.Infof("%s created", alertmanagerConfigPath)
 }
 
+// collectRuleTemplates collects IA rule templates from various sources like
+// templates shipped with PMM and defined by the users.
+func (svc *Service) collectRuleTemplates() {
+	shippedFilePaths, err := filepath.Glob(svc.shippedRuleTemplatePath)
+	if err != nil {
+		svc.l.Errorf("Failed to get paths of template files shipped with PMM: %s.", err)
+		return // keep previously loaded rules
+	}
+
+	for _, path := range shippedFilePaths {
+		rules, err := svc.loadRuleTemplates(path)
+		if err != nil {
+			svc.l.Errorf("Failed to load shipped rule template file: %s, reason: %s.", path, err)
+			return // keep previously loaded rules
```

Comment:
Well, if one file was loaded, but another wasn't, we don't keep previously loaded rules – we already collected updateRules below.

And it seems we never remove rules?